lib/MusicBrainz/Server/Data/Area.pm

```diff
@@ -21,17 +21,17 @@ use MusicBrainz::Server::Data::Utils qw(
 use MusicBrainz::Server::Data::Utils::Cleanup qw( used_in_relationship );
 
 extends 'MusicBrainz::Server::Data::Entity';
-with 'MusicBrainz::Server::Data::Role::Relatable';
-with 'MusicBrainz::Server::Data::Role::Name';
-with 'MusicBrainz::Server::Data::Role::Annotation' => { type => 'area' };
-with 'MusicBrainz::Server::Data::Role::Alias' => { type => 'area' };
-with 'MusicBrainz::Server::Data::Role::GIDEntityCache';
-with 'MusicBrainz::Server::Data::Role::DeleteAndLog' => { type => 'area' };
-with 'MusicBrainz::Server::Data::Role::PendingEdits' => { table => 'area' };
-with 'MusicBrainz::Server::Data::Role::Merge';
-with 'MusicBrainz::Server::Data::Role::LinksToEdit' => { table => 'area' };
-with 'MusicBrainz::Server::Data::Role::Tag' => { type => 'area' };
-with 'MusicBrainz::Server::Data::Role::Collection';
+with 'MusicBrainz::Server::Data::Role::Alias' => { type => 'area' },
```

Comment:
I understand that the whole point of combining them all in one `with` is that the order doesn't matter anymore since they're all evaluated together and crash if they clash (like in the first commit here).

Reply:
The citation also says "You should never assume any ordering", but I understand the fear of us actually assuming an ordering. I would like to think that if all tests pass it's probably ok, but that might be putting too much trust on our tests?

Reply:
Right...they *should* be written in a way that it doesn't matter, but I'm worried about introducing some subtle bug that our tests don't catch.

If tests do pass that is reassuring, and maybe we don't have that many uses of method modifiers under `Data::` that we could actually review them all.

Reply:
I was going to say there there's at least one case where the order affects behavior (though it doesn't cause an error or exception): consuming `Data::Role::Relatable` after `Data::Role::GIDEntityCache` means redirected gids won't be cached.

However, I discovered there's a bug introduced by 0c48badf249a09eff6ddce7166c9d5b450cd7150 where it doesn't cache the redirected `gid` in any case.

But I do think this demonstrates that there may be subtle differences in behavior we cannot entirely predict, and would prefer keeping the original order.

Reply:
Ok, I guess I can put everything back in the previous order :) I suspect it's random in *most* cases, but some might be intentional.

Are you sending a PR for that bug?

Reply:
Made the changes, anyway - will do the same for the subsequent PRs later.

Reply:
Thanks!

> Are you sending a PR for that bug?

I'll try to look into it this week at least.